cryptography/src/ed25519.rs

```diff
@@ -125,6 +125,47 @@ impl Scheme for Ed25519 {
     }
 }
 
+impl Ed25519 {
+    /// Check each signature is valid for the given message and public key.
+    ///
+    /// Because namespace is prepended to message before signing, the namespace
+    /// provided here must match the namespace provided during signing.
+    pub fn batch_verify(inputs: Vec<SignatureVerificationInput>) -> bool {
```

Comment:
To make this more memory efficient, I recommend that we just wrap the `batch` with a new struct (like `Ed25519Batch` or something) and allow inputs to be added one at a time.

Otherwise, I imagine that the caller will create a vector just for that vector to be converted into another vector/data structure. I've been thinking about doing the same for the bls "multiple" stuff.